tests/protocol/pool/Pool.Liquidations.Rwa.t.sol

```diff
@@ -0,0 +1,959 @@
+import {LiquidationLogic} from 'src/contracts/protocol/libraries/logic/LiquidationLogic.sol';
+import {IERC20Detailed} from 'src/contracts/dependencies/openzeppelin/contracts/IERC20Detailed.sol';
+import {PercentageMath} from 'src/contracts/protocol/libraries/math/PercentageMath.sol';
+import {WadRayMath} from 'src/contracts/protocol/libraries/math/WadRayMath.sol';
+import {DataTypes} from 'src/contracts/protocol/libraries/types/DataTypes.sol';
+import {Errors} from 'src/contracts/protocol/libraries/helpers/Errors.sol';
+import {AggregatorInterface} from 'src/contracts/dependencies/chainlink/AggregatorInterface.sol';
+import {TestnetRWAERC20} from 'src/contracts/mocks/testnet-helpers/TestnetRWAERC20.sol';
+import {RwaAToken} from 'src/contracts/protocol/tokenization/RwaAToken.sol';
+import {LiquidationDataProvider} from 'src/contracts/helpers/LiquidationDataProvider.sol';
+import {TestnetProcedures} from 'tests/utils/TestnetProcedures.sol';
+
+contract PoolLiquidationsRwaTests is TestnetProcedures {
+  using UserConfiguration for DataTypes.UserConfigurationMap;
+
+  struct RwaTokenInfo {
+    address rwaToken;
+    address rwaAToken;
+    address user;
+    address liquidator;
+  }
+
+  enum LiquidationType {
+    Partial,
+    Full // Full means the maximum liquidatable amount (even if that is half of the position)
+  }
+
+  /// @dev if priceImpactToken is non-zero, then its price must be changed
+  /// such that the healthFactorTarget is reached. otherwise, time must be
+  /// skipped such that the target is reached through borrow interest accrued.
+  struct LiquidationCheck {
+    address user;
+    address supplyToken;
+    uint256 supplyAmount;
+    address borrowToken;
+    address priceImpactToken;
+    uint256 healthFactorTarget;
+    LiquidationType liquidationType;
+    bool receiveAToken;
+    address liquidator;
+    bytes expectedRevertData;
+    bool expectFullLiquidation;
+    bytes beforeLiquidationCallbackCalldata;
+  }
+
+  struct CheckLiquidationVars {
+    uint256 borrowAmount;
+    uint256 avgLiqThreshold;
+    uint256 totalCollateralInBaseCurrency;
+    uint256 totalDebtInBaseCurrency;
+    int256 priceImpactPercent;
+    uint256 timeToSkip;
+    uint256 totalCollateralInBaseCurrencyTarget;
+    uint256 totalDebtInBaseCurrencyTarget;
+    uint256 liquidationAmount;
+  }
+
+  address internal david;
+  RwaTokenInfo[] internal rwaTokenInfos;
+  LiquidationDataProvider internal liquidationDataProvider;
+
+  function setUp() public {
+    initTestEnvironment();
+
+    liquidationDataProvider = new LiquidationDataProvider(
+      address(contracts.poolProxy),
+      address(contracts.poolAddressesProvider)
+    );
+
+    address buidlLiquidator = makeAddr('BUIDL_LIQUIDATOR_1');
+    address wtgxxLiquidator = makeAddr('WTGXX_LIQUIDATOR_1');
+    address ustbLiquidator = makeAddr('USTB_LIQUIDATOR_1');
+
+    vm.startPrank(poolAdmin);
+    // alice is the only authorized BUIDL holder amongst the users
```

Comment:
should we just authorize here and not in TestnetProcedures (so de-authorizations won't be needed)? seems a bit hidden there

Reply:
authorization takes place only if `_initTestEnvironment` is called with `mintUserTokens`. Here I think it's actually easier to de-authorize instead of authorize, mint and approve pool. wdyt?

Reply:
[0fc2dad](https://github.com/aave/aave-v3-horizon/pull/1/commits/0fc2dad326889c78fa2cd968375e0ef98f12989a)